### DISCUSSION TITLE
Lights react to audio and Bluetooth connection (party mode)

### DISCUSSION THREAD

**Discussion by photoshy** [gpstar81/GPStar-proton-pack#362] [Ideas]
I think it could be cool to have a party mode where the cyclotron and power cell lights could act as a visualiser (maybe also Bluetooth connectivity so the pack can play music off a phone)  and have the lights pulse with the music

**Comment by nomakewan** [comment]
This cannot be done (caveat: with our existing hardware).

Reason being, we have no way for the GPStar Pack PCB (which handles all the LEDs etc) to receive data about audio waveforms (which would be required to be able to have any hope of altering LED states along with music).

To clarify, the polyphonic audio boards are totally independent of the main processor. They play audio on request, and can send back track information (such as how many tracks exist on the SD card and if a specific track is currently being played), but cannot send any information about the actual audio waveforms themselves. Likewise, we have no microphone input which could pick up on ambient sounds and react to them. Nor do we have a bluetooth A2DP client.
